orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/config/RedisOrcaQueueConfiguration.kt

```diff
@@ -41,13 +44,16 @@ import java.util.*
 @EnableConfigurationProperties(ObjectMapperSubtypeProperties::class)
 class RedisOrcaQueueConfiguration : RedisQueueConfiguration() {
 
-  @Autowired fun redisQueueObjectMapper(mapper: ObjectMapper,
-                                             objectMapperSubtypeProperties: ObjectMapperSubtypeProperties) {
+  @Autowired
+  fun redisQueueObjectMapper(mapper: ObjectMapper,
+                             objectMapperSubtypeProperties: ObjectMapperSubtypeProperties,
+                             taskResolver: TaskResolver) {
     mapper.apply {
       registerModule(KotlinModule())
       registerModule(
         SimpleModule()
           .addDeserializer(ExecutionType::class.java, ExecutionTypeDeserializer())
+          .addDeserializer(Class::class.java, TaskTypeDeserializer(taskResolver))
```

Comment:
Did you mean to use `Class` here? Is this right?

Reply:
It was working with this but I need to play around a bit more as it does look a bit strange.